lighthouse/helpers/plates.py

```diff
@@ -134,10 +133,27 @@ def get_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
 
 
 def get_positive_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
-    query_filter = copy.deepcopy(POSITIVE_SAMPLES_MONGODB_FILTER)
-    query_filter[FIELD_PLATE_BARCODE] = plate_barcode
+    """Get a list of documents which correspond to filtered positive samples for a specific plate.
 
-    samples_for_barcode = find_samples(query_filter)
+    Args:
+        plate_barcode (str): the barcode of the plate to get samples for.
+
+    Returns:
+        Optional[List[Dict[str, Any]]]: the list of samples for this plate.
+    """
+    samples_collection = app.data.driver.db.samples
+
+    # The pipeline defines stages which execute in sequence
+    pipeline = [
+        # 1. First run the positive match stage
+        STAGE_MATCH_POSITIVE,
+        # 2. We are only interested in the samples for a particular plate
+        {"$match": {FIELD_PLATE_BARCODE: plate_barcode}},
+    ]
```

Comment:
should this be the other way? first match the barcode and then match the positive? it should be quicker:

https://docs.mongodb.com/manual/core/aggregation-pipeline-optimization/

Reply:
Yes, good spot!